modules/rhcos-setting-grub-password-day-2.adoc

```diff
@@ -0,0 +1,144 @@
+Enter password:
+Reenter password:
+PBKDF2 hash of your password is
+grub.pbkdf2.sha512.10000.5AE6255...
+----
+
+. Create a script that validates your configurations and sets the GRUB config for the root user with the newly created password:
++
+.rhcos_set_grub_password.sh
+[source,bash]
+----
+set -euo pipefail
+
+STAMPFILE="/var/lib/.grub_password_<date_changed>"
+GRUBPASSWORD="grub.pbkdf2.sha512.10000.58E6255..."
+
+main () {
+
+# Validate GRUB config
+  if [[ $(grep -c 'source $prefix/user.cfg' '/boot/grub2/grub.cfg') != 1]]; then
+    echo "Invalid GRUB config: Missing user.cfg"
+    return 1
+  fi
+
+# Validate ostree repo config
+  if [[ $(grep -c 'bls-append-except-default=grub_users=""' '/sysroot/ostree/repo/config') != 1 ]]; then
+    echo "Invalid ostree repo config: Missing bls-append-except-default"
+    return 1
+  fi
+
+# Remount /boot as RW
+  mount -o remount,rw /boot
+
+# Write config
+  echo -e "set superusers="root"\npassword_pbkdf2 root {GRUBPASSWORD}" > /boot/grub2/user.cfg
+  chmod 400 /boot/grub2/user.cfg
+
+# Write stamp file
+  touch "{STAMPFILE}"
+
+}
+----
+
+. Create a systemd service unit that launches the script on boot:
++
+.rhcos-set-grub-password.service
+[source,text]
+----
+[Unit]
+Description=Set GRUB password
+ConditionPathExists=!/var/lib/.grub_password_<date_changed>
+RequiresMountsFor=/boot
+
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+MountFlags=slave
+ExecStart=/usr/local/bin/rhcos_set_grub_password.sh
+
+[Install]
+WantedBy=multi-user.target
+----
+
+. Create a Butane config that includes the files from the previous steps:
++
+.40-worker-custom-journald.bu
+[source,yaml]
+----
+variant: openshift
+version: {product-version}.0
+metadata:
+  name: 99-worker-grub-password
+  labels: machineconfiguration.openshift.io/role: worker
+storage:
+  files:
+    - path: /usr/local/bin/rhcos_set_grub_password.sh
+      mode: 0700
+      overwrite: true
+      contents:
+        inline: |
+        ... <1>
+
+    - path: /etc/systemd/system/rhcos-set-grub-password.service
+      mode: 0644
+      overwrite: true
+      contents:
+        inline: |
+        ... <2>
+----
+<1> The contents of the file `rhcos_set_grub_password.sh`.
+<2> The contents of the file `rhcos-set-grub-password.service`.
+
+. Generate a `MachineConfig` object, `40-worker-custom-journald.yaml`, that contains the configuration you want to deliver to the nodes:
++
+[source,terminal]
+----
+$ butane 40-worker-custom-journald.bu -o 40-worker-custom-journald.yaml
+----
+
+. Apply the machine config:
++
+[source,terminal]
+----
+$ oc apply -f 40-worker-custom-journald.yaml
+----
++
+The controller detects the new `MachineConfig` object and generated a new `rendered-worker-<hash>` version.
+
+. Monitor the status of the rollout of the new rendered configuration to each node:
++
+[source,terminal]
+----
+$ oc describe machineconfigpool/worker
+----
++
+Once the password resets after a node reboot, the machine config can be removed.
```

Comment:
This line is a bit confusing. Do we want to say that applied MachineConfig can/should be removed once grub password has been set? In that case, newly scaled-up node won't have those grub config applied.

Reply:
Oh, that's a good point. We need to keep it in.